source/core/module_coreHelpers.js

```diff
@@ -75,33 +104,68 @@
                     str += String.fromCharCode(u0);
                     continue;
                 }
+
+                // Look ahead to validate the UTF-8 structure
+                if ((u0 & 0xE0) === 0xC0) {
```

Comment:
If I may push back on that a little, these constants don't have any semantic meaning that would lend themselves to good named constants; they are defined in such a way that the encoding space for UTF-8 keeps more common characters with more compact encodings then less common ones.   If you are trying to work through this code to see how it encodes a particular byte array portion, having the hex constants directly in the code actually makes it easier to understand/debug, not harder, because there is no meaning you would add by changing the values to symbolic constants.